toolchain/lower/function_context.cpp

```diff
@@ -51,6 +51,32 @@ auto FunctionContext::LowerBlock(SemIR::InstBlockId block_id) -> void {
   }
 }
 
+// Handles typed instructions for LowerInst. We don't use LowerFunctionInst
+// overloads for the "if" cases because it's hard to get the overload behavior
```

Comment:
```suggestion
// overloads for instructions that are not lowered, are always constant, or are always types because it's hard to get the overload behavior
```

Reply:
Rephrased the comment to try to avoid talking about specific cases:

```
// Handles typed instructions for LowerInst. Many instructions lower using
// HandleInst, but others are unsupported or have trivial lowering.
//
// A different approach would be to have the logic below instead done through
// HandleInst overloads. However, forward declarations of HandleInst exist for
// all `InstT` types, which would make getting the right overload resolution
// complex.
```